From 011b2dabf0b2003d06e8ae60401b2db58808eede Mon Sep 17 00:00:00 2001
From: liubo <liubo254@huawei.com>
Date: Tue, 20 Sep 2022 21:10:47 +0800
Subject: [PATCH] add ETMEM feature CONFIG to mm/Kconfig
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 011b2dabf0b2003d06e8ae60401b2db58808eede
Modified-by-SEL: No


euleros inclusion
category: feature
feature: etmem
bugzilla: https://gitee.com/openeuler/kernel/issues/I5DC4A

-------------------------------------------------
etmem, the memory vertical expansion technology,
uses DRAM and high-performance storage new media to form multi-level
memory storage.

The etmem feature was introduced in the previous commit
(aa7f1d222cdab88f12e6d889437fed6571dec824),but only the config
options for the etmem_swap and etmem_scan modules were added,
and the config options for the etmem feature were not added,
so in this commit, the CONFIG_ETMEM option for
the etmem feature was added

Signed-off-by: liubo <liubo254@huawei.com>
Reviewed-by: Miaohe Lin <linmiaohe@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/Kconfig | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/mm/Kconfig b/mm/Kconfig
index 9e66dfb15c52..8ff588380c55 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -514,18 +514,28 @@ config MEMCG_QOS
 
 config ETMEM_SCAN
 	tristate "module: etmem page scan for etmem support"
-        depends on MMU
-        depends on X86 || ARM64
+	depends on ETMEM
 	help
 	 etmem page scan feature
 	 used to scan the virtual address of the target process
 
 config ETMEM_SWAP
 	tristate "module: etmem page swap for etmem support"
+	depends on ETMEM
+	help
+	 etmem page swap feature
+
+config ETMEM
+	bool "Enable etmem feature"
 	depends on MMU
 	depends on X86 || ARM64
+	default n
 	help
-	 etmem page swap feature
+	  etmem is a tiered memory extension technology that uses DRAM and memory
+	  compression/high-performance storage media to form tiered memory storage.
+	  Memory data is tiered, and cold data is migrated from memory media to
+	  high-performance storage media to release memory space and reduce
+	  memory costs.
 
 config USERSWAP
 	bool "Enable User Swap"
-- 
2.34.1

